docs/development/design/organizations.rst

```diff
@@ -0,0 +1,148 @@
+-------------------------------
+
+Since we are not replacing the current implementation,
+we don't need to migrate current projects from the community site nor from the corporate site.
+
+How to migrate the organizations app
+------------------------------------
+
+The migration can be split in:
+
+#. Remove/simplify code from the organizations app on the corporate site.
+#. Isolate/separate models and code that isn't going to be moved.
+#. Start by moving the models, managers, and figure out how to handle migrations.
+#. Move the rest of the code as needed.
+#. Activate organizations app on the community site.
+#. Integrate the code from the community site to the new code.
+
+We should start by removing unused features and dead code from the organizations in the corporate site,
+and simplify existing code if possible (some of this was already done).
+Another simplification that can be done is:
+make the relationship between the models ``Organization`` and ``Project`` one to many
+(currently many to many).
+
+Isolate/separate the models to be moved from the ones that aren't going to be moved.
+We should move the models that aren't going to me moved to another app.
+
+- Plan
+- PlanFeature
+- Subscription
+
+This app can be named *subscriptions*.
+We can get around the table names and migrations by setting the explicitly the table name to ``organizations_<model>``,
+and doing a fake migration.
+Following suggestions in https://stackoverflow.com/questions/48860227/moving-multiple-models-from-one-django-app-to-another.
+Code related to subscriptions should be moved out from the organizations app.
+
+After that, it should be easier to move the organizations *app* (or part of it)
+to the community site (and no changes to table names would be required).
+
+We start by moving the models.
+
+- Organization
+- OrganizationOwner
+- Team
+- TeamInvite
+- TeamMember
+
+Migrations aren't moved, since all current migrations depend on other models that aren't
+going to be moved.
+In the community site we run an initial migration,
+for the corporate site we run a fake migration.
+
+For managers and querysets that depend on subscriptions,
+we can use our pattern to make overridable classes (inheriting from ``SettingsOverrideObject``).
+
+Templates, urls, views, forms, notifications, signals, tasks can be moved later
+(we just need to make use of the models from the ``readthedocs.organizations`` module).
+
+If we decide to integrate organizations in the community site,
+we can enable the app.
+
+After the app is moved,
+we can move more code that depends on organizations to the community site.
+
+Namespace
+---------
+
+Currently we use the project's slug as namespace,
+in the commercial site we use the combination of ``organization.slug`` + ``project.slug`` as namespace.
+
+For the community site probably this approach isn't the best,
+since we always serve docs publicly from ``slug.readthedocs.io``.
+And most of the users don't have a custom domain.
```

Comment:
What is our plan for slugs for organization projects in community then? Are we not using the organization slug at all? I have concerns about supporting both patterns at once. A collision is introduced if you slug a project with slug `foo-something` and an organization project for organization slug `foo` and project name `something` -- ie both produce the same slug.

Reply:
I'm suggesting to use the project slug for users and orgs for .org. So we have `foo-something` and `something`.

Reply:
Lie if a user from the pallets org creates a project named flask, the slug would be `flask`, no `pallets-flask`.